### DISCUSSION TITLE
[Feature] Identify popular faces

### DISCUSSION THREAD

**Discussion by sujaldev** [immich-app/immich#6078] [Feature Request]
### The feature

Would it be possible to identify popular faces and then display them in a separate section in the the clients on the people page? I have no clue how it'd work but it'd be pretty cool to have this.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by kingvuke** [comment]
How do you define "popular faces"?

**Reply by sujaldev** [reply]
I don't think we'd have to define "popular faces" in the sense that an algorithm would automatically find such people, as I assume that the implementation of this feature would require manually feeding in a list of people who are considered famous by someone involved in the creation of this list, along with enough images of the faces proposed in the list. From a quick search there seem to be plenty of databases available for celebrities but that probably excludes people like say famous mathematicians and the like.

**Comment by aisbergde** [comment]
let's say: if face tagging would be distinguish between face detection and face assignment (I think, this is on the road map), then it could be possible to use face vector libraries. I don't know, if these libraries exists. They should be based on the same model, I think. And I am talking only about the technical aspects, not about privacy rules.

A related point is to share face definition between immich users, to have immich face model specific databases to share. To share on the same instance or to share between different immich instances. Or just to have a "knowledge backup" about detected persons. Because currently we will lost all clustering and assignments, if the model is switched.

It looks like currently each immich user need to cluster and name their own assets. But we have "popular people", we want to share their clusters and assignments: Mama, Papa, Oma, Opa, children, ...

And it would be helpful to be able to share them.

**Comment by jaybinks** [comment]
I think this could be as simple as, when visiting /people show a count of how many times this face has been identified (and sort by most common).

That way you could scroll and go ... ooh there is a common face ive not "named"

**Reply by alextran1502** [reply]
@jaybinks there are already sorted by faces with name then faces with the most assets.

**Reply by jaybinks** [reply]
Thanks... that wasnt obvious to me.

**Reply by ireun** [reply]
So I guess we could close this discussion? "popular faces" are at the beginning of `/people/`

**Reply by bo0tzz** [reply]
I believe the request is for identifying celebrities, not the most-photographed people on the instance.

**Comment by ghost** [comment]
Hi everyone! I arrived here after my duplicate request was closed, and I'm really glad this discussion is active again.

To add a real-world use case to what @bo0tzz and @sujaldev mentioned: I am a Scenography student, and due to my Art History studies, my Immich gallery is filled with photos of paintings, sculptures, and historical figures. Having Immich automatically recognize and tag historical personalities (like Napoleon, Julius Caesar, or even faces in famous paintings) alongside standard personal contacts would be an absolute game-changer for visual research and archiving.

I really like @aisbergde's idea regarding "face vector libraries". Instead of training a new AI, we could generate the vectors for these historical figures using Immich's current ML model and allow users to import this specific database.

Since Immich is open source, I would love to try and contribute to this feature. I am willing to start investigating the technical feasibility of generating these custom "historical face vectors" and figuring out how they could be imported into the database to match against user photos.

@bo0tzz or anyone from the core team: would a PR introducing the ability to import external face vector libraries be something you'd be open to? If so, any quick pointers on which files in the ML container / Backend I should start looking at to understand the current face clustering logic?

I always apologize for my unexceptional English

**Reply by ghost** [reply]
Just to add a quick follow-up thought to my previous comment regarding the "popular faces" database:

I realized that if this feature were to rely on a single, pre-packaged global library, there is a high risk it would be heavily biased towards modern US/UK celebrities and pop culture. It would likely leave out niche, international, or historical figures (like the ones I need for my Art History studies).

To avoid this cultural bias and make the feature truly universal, the best approach might be a modular one. Instead of Immich shipping with a built-in list of famous faces, it could simply provide the mechanism to import custom face vector libraries.

This way, the community could create and share specific thematic packs (e.g., "Italian Renaissance figures", "Roman Emperors", "1920s Cinema", etc.) and users could choose to install only the ones relevant to their own archives.

**Reply by bo0tzz** [reply]
I think what data to use is the main issue here, and it seems worth seeing what the options there are before working out whether it's worth including any of those in Immich somehow.

**Reply by ghost** [reply]
That makes total sense. You are absolutely right: building the implementation is secondary to actually having good, usable data.

I can take the lead on the research phase for my specific use case. I'll start looking into what open-source, public domain datasets (or academic art history databases) currently exist for historical figures and classical art portraits (e.g., pulling from Wikimedia Commons, museum Open APIs, or existing academic ML projects).

I will gather some options and report back here with what I find. That way, we can evaluate the quality and format of the available data before deciding if it's worth integrating into Immich.
